docs/components.md

```diff
@@ -0,0 +1,63 @@
+# Components
+
+FMN consists of several components, most of which run in Fedora Infrastructure.
+
+![Components of FMN](images/fmn-components.png)
+
+## Message Consumer
+
+The *Message Consumer* reads messages carried on the *Message Bus*. If a message is matched by a
+rule, it triggers the appropriate [*Notification Sender*](#notification-sender) to send an email or
+chat message to the user who set up the rule.
+
+It queries *FASJSON* and *dist-git (Pagure)* for information about users, groups and projects and
+stores the information in the [*Shared Service Cache*](#shared-service-cache) (which is also used by
+the [*Configuration Backend*](#configuration-backend)) and is responsible for invalidating cached
+entries when receiving messages that affected objects have changed.
+
+## Notification Sender
+
+A *Notification Sender* receives messages over a private message queue from the [*Message
+Consumer*](#message-consumer), triggering it to send notifications to users over various
+communication channels, such as IRC, email or Matrix.
```

Comment:
I would add: "There can be only one instance of the IRC and Matrix senders". I would also add that the sender does not request information from the rest of the Fedora Infrastructure (no requests to dist-git or FASJSON). All the necessary information is already contained in the message it receives from the Message Consumer.